CODEBASE_ANALYSIS.md

```
@@ -0,0 +1,193 @@
+└─────────────────────────────────────────────────────────────────┘
+                              │
+                              ▼
+┌─────────────────────────────────────────────────────────────────┐
+│                         core/                                    │
+│  ┌────────────┐  ┌────────────┐  ┌────────────┐  ┌──────────┐ │
+│  │  config    │  │  database  │  │exec_context│  │observability│
+│  │(env vars)  │  │(Postgres)  │  │(contextvars)│ │(Phoenix)  │ │
+│  └────────────┘  └────────────┘  └────────────┘  └──────────┘ │
+└─────────────────────────────────────────────────────────────────┘
+```
+
```

markdownlint-cli2 (0.21.0)

[warning] Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)